docs/concepts/registries.md

```diff
@@ -28,32 +28,27 @@ public ExampleMod() {
 }
 ```
 
-### `Register` events
+### `RegisterEvent`
 
-The `RegistryEvent`s are the second and more flexible way to register objects. These [events][] are fired after the mod constructors and before the loading of configs.
-
-The event used to register objects is `RegistryEvent$Register<T>`. The type parameter `T` should be set to the type of the object being registered. Calling `#getRegistry` will return the registry, upon which objects are registered with `#register` or `#registerAll`. 
+`RegisterEvent` is the second way to register objects. This [event] is fired for each registry after the mod constructors and before the loading of configs. Objects are registered using `#register` by passing in the registry key, the name of the registry object, and the object itself.
 
 Here is an example: (the event handler is registered on the *mod event bus*)
 
 ```java
 @SubscribeEvent
-public void registerBlocks(RegistryEvent.Register<Block> event) {
-  event.getRegistry().registerAll(new Block(...), new Block(...), ...);
+public void register(RegisterEvent event) {
+  event.register(ForgeRegistries.Keys.BLOCKS, new ResourceLocation(MODID, "rock"), () -> new Block(...));
```

Comment:
This will create a new resource location for every single forge and vanilla registry as they will all invoke this event handler. Perhaps use the helper variant and recommend people use that instead?